apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableReadRowsRequestManager.cs

```diff
@@ -0,0 +1,187 @@
+
+            {
+                TableNameAsTableName = _originalRequest.TableNameAsTableName,
+            };
+
+            #region Set up RowFilter
+
+            if (_originalRequest.Filter != null)
+            {
+                newReadRowsRequest.Filter = new RowFilter();
+                newReadRowsRequest.Filter = _originalRequest.Filter;
+            }
+
+            #endregion
+
+            #region Set up RowsLimit
+
+            // If the row limit is set, update it.
+            long numRowsLimit = _originalRequest.RowsLimit;
+
+            // Update <see cref="numRowsLimit"/> by removing the number of rows already read.
+            if ((numRowsLimit -= RowCount) > 0)
+            // The remaining number of rows must be greater than 0.
+            {
+                newReadRowsRequest.RowsLimit = numRowsLimit;
+            }
+
+            #endregion
+
+            #region Set up RowSet
+
+            newReadRowsRequest.Rows = new RowSet();
+            newReadRowsRequest.Rows = FilterRows();
+
+            #endregion
+
+            return newReadRowsRequest;
+        }
+
+        private RowSet FilterRows()
+        {
+            RowSet originalRows = _originalRequest.Rows;
+
+            if (LastFoundKey == null)
+            {
+                return originalRows;
+            }
+
+            RowSet newRowSet = new RowSet();
+
+            foreach (ByteString key in originalRows.RowKeys)
+            {
+                if (!StartKeyIsAlreadyRead(key))
+                {
+                    newRowSet.RowKeys.Add(key);
+                }
+            }
+
+            foreach (RowRange rowRange in originalRows.RowRanges)
+            {
+                RowRange.EndKeyOneofCase endKeyOneofCase = rowRange.EndKeyCase;
+
+                if (endKeyOneofCase == RowRange.EndKeyOneofCase.EndKeyClosed
+                    && EndKeyIsAlreadyRead(rowRange.EndKeyClosed)
+                    || endKeyOneofCase == RowRange.EndKeyOneofCase.EndKeyOpen
+                    && EndKeyIsAlreadyRead(rowRange.EndKeyOpen))
+                {
+                    continue;
+                }
+
+                RowRange newRange = rowRange;
+                RowRange.StartKeyOneofCase startKeyOneofCase = rowRange.StartKeyCase;
+
+                if (startKeyOneofCase == RowRange.StartKeyOneofCase.StartKeyClosed
+                    && StartKeyIsAlreadyRead(rowRange.StartKeyClosed)
+                    || (startKeyOneofCase == RowRange.StartKeyOneofCase.StartKeyOpen
+                        && StartKeyIsAlreadyRead(rowRange.StartKeyOpen))
+                    || startKeyOneofCase == RowRange.StartKeyOneofCase.None)
+                {
+                    newRange.StartKeyOpen = LastFoundKey;
+                }
+                newRowSet.RowRanges.AddRange(new List<RowRange> { newRange });
+            }
+
+            return newRowSet;
+        }
+
+        private bool StartKeyIsAlreadyRead(ByteString startKey)
```

Comment:
Make parameter `BigtableByteString` and then the code below can be:
```csharp
... => LastFoundKey != null && (startKey != null || startKey <= LastFoundKey);
```
(similar for `EndKeyIsAlreadyRead` below).